### ISSUE TITLE
Python doesn't raise errors if the C++ compilation fails

### ISSUE THREAD

**Issue by WetzelVictor** [pyphs/pyphs#101]
Hi PyPHS users!

I've been having troubles, related to the issue #99.

### Summary
If an error occurs while running the `run.sh` for `C++` compilation, Python doesn't raise any error and continues running. 
Moreover, It displays older simulations results (obtained with the Python simulation method) if they exists, which could make you believe that it worked.

--------------------------------------
##### Informations
* `pyphs` version: master branch on GitHub
* `cmake` version: 3.5.1
* `Ubuntu` version: 16.04.4 LTS Xenial Xerus

I took the same example as the one described on the issue #99, and specified a wrong `cmake` path to make the compilation fail:
```python
import sys
sys.path.append('/home/victorw/anaconda/envs/dev_pyphs/workspace/lib_pyphs_dev/')
from pyphs import Core    # import the pyphs.Core class

label = 'dampedosc'       # system's label
core = Core(label)        # instanciate a Core object

xm = core.symbols('xm')    # mass state symbol
m = core.symbols('m')      # mass parameter symbol
hm = xm**2/(2.*m)          # kinetic energy expression
core.add_storages(xm, hm)  # add mass to Core

xk, k = core.symbols(['xk', 'K0']) # stiffness symbols
hk = k*xk**2/2.                    # potential energy expression
core.add_storages(xk, hk)          # add stiffness to Core 


wa, a = core.symbols(['w_a', 'a'])       # damper symbols
za = a*wa                                # damper expression
core.add_dissipations(wa, za)            # add damper to core

core.set_Jxx([[0, -1],
             [+1, 0]])
core.set_Jxw([[-1], 
             [0]])    

# ==================================
# ======= VALUES ===================
m_value = 0.1                         # mass value (kg)
F = 5e2                               # natural frequency (HZ)
k_value = m_value*(2*3.14157*F)**2    # stiffness value (N/m)
a_value = 1.                          # damper value (N.s/m)

mysubs = {m: m_value, 
        k: k_value,
        a: a_value}
core.subs.update(mysubs)                


# ==================================
# ========== SIMULATIONS ===========
import os                                
config = {'fs': 1e4,                                 # sample rate (Hz)
          'path': os.path.join(os.getcwd(), label),   # folder for output results
          'lang':'c++',
          'cmake':'usr/cmake' # <<<<<<<<<< It will fail on my machine
         }
T = 1.                                               # simulation time (s)
nt = int(T*config['fs'])                             # number of time steps (int)

simu = core.to_simulation(config=config)   # `Simulation` object.
simu.method.x                              # get state ordering


inits = {'x': [0.1, 0.]}                  # initial values
simu.init(nt=nt, inits=inits)             # initialize `Data` object

simu.process()                            # process
```
This script doesn't show any errors.

-------------------------------------
### Fix
This [PR](https://github.com/pyphs/pyphs/pull/102) #102 proposes a fix, based on a temp file. Everything is explained, step by step.

With this fix, Python raises this error:
```python
Traceback (most recent call last):
  File "damped_osc.py", line 58, in <module>
    simu.process()                            # process
  File "/home/victorw/anaconda/envs/dev_pyphs/workspace/lib_pyphs_dev/pyphs/numerics/simulations/simulation.py", line 261, in process
    self._process_cpp()
  File "/home/victorw/anaconda/envs/dev_pyphs/workspace/lib_pyphs_dev/pyphs/numerics/simulations/simulation.py", line 367, in _process_cpp
    self._check_cpp_runtime_errors()
  File "/home/victorw/anaconda/envs/dev_pyphs/workspace/lib_pyphs_dev/pyphs/numerics/simulations/simulation.py", line 388, in _check_cpp_runtime_errors
    + os.path.join(self.cpp_path, stdError[-1]))
RuntimeError: Bash/Cmake error while trying to compile cpp simulation: see file /home/victorw/anaconda/envs/dev_pyphs/workspace/dampedosc/dampedosc/stderr: step 2 - Cmake Build
```

---------------------------------------
### Thoughts
I'm really not sure this is the best way of fixing it, but I tested it a little, and it performs well and stops the execution of the Python script if something goes wrong.

We can discuss implementation on the [PR](https://github.com/pyphs/pyphs/pull/102) #102. I raised more questions about this.

Thank you for your insights on this matter :)

Victor W.

**Comment by WetzelVictor** [comment]
Issue closed with PR #102 and merge #104

**Comment by afalaize** [comment]
We will close this issue when passed into master branch
